tutorials/mesh_generation/4_tree_mesh.py

```diff
@@ -0,0 +1,254 @@
+"""
+Tree Meshes
+===========
+
+Compared to tensor meshes, tree meshes are able to provide higher levels
+of discretization in certain regions while reducing the total number of
+cells. Tree meshes belong to the class (:class:`~discretize.TreeMesh`).
+Tree meshes can be defined in 2 or 3 dimensions. Here we demonstrate:
+
+    - How to create basic tree meshes in 2D and 3D
+    - Strategies for local mesh refinement
+    - How to plot tree meshes
+    - How to extract properties from tree meshes
+
+To create a tree mesh, we first define the base tensor mesh (a mesh
+comprised entirely of the smallest cells). Next we choose the level of
+discretization around certain points or within certain regions. When
+creating tree meshes, we must remember certain rules:
+
+    - The number of base mesh cells in x, y and z must all be powers of 2
+    - We cannot refine the mesh to create cells smaller than those defining the base mesh
+    - The range of cell sizes in the tree mesh depends on the number of base mesh cells in x, y and z
+    
+
+"""
+
+###############################################
+#
+# Import Packages
+# ---------------
+#
+# Here we import the packages required for this tutorial.
+#
+
+from discretize import TreeMesh
+from discretize.utils import matutils, meshutils
+import matplotlib.pyplot as plt
+import numpy as np
+
+# sphinx_gallery_thumbnail_number = 4
+
+###############################################
+# Basic Example
+# -------------
+#
+# Here we demonstrate the basic two step process for creating a 2D tree mesh
+# (QuadTree mesh). The region of highest discretization if defined within a
+# rectangular box. We use the keyword argument *octree_levels* to define the
+# rate of cell width increase outside the box.
+#
+
+dh = 5    # minimum cell width (base mesh cell width)
+nbc = 64  # number of base mesh cells in x
+
+# Define base mesh (domain and finest discretization)
+h = dh*np.ones(nbc)
+mesh = TreeMesh([h, h])
+
+# Define corner points for rectangular box
+xp, yp = np.meshgrid([120., 240.], [80., 160.])
+xy = np.c_[matutils.mkvc(xp), matutils.mkvc(yp)]  # matutils.mkvc creates vectors
```

Comment:
```suggestion
xy = np.c_[mkvc(xp), mkvc(yp)]  # mkvc creates vectors
```

Reply:
etc in this script. This will simplify maintenance down the road if we do some re-organization of the utils